src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -2012,7 +2012,7 @@ ReadFromMergeTree::AnalysisResultPtr ReadFromMergeTree::selectRangesToRead(
     bool add_index_stat_row_for_pk_expand = false;
 
     {
-        MergeTreeDataSelectExecutor::filterPartsByPartition(
+        auto res = MergeTreeDataSelectExecutor::filterPartsByPartition(
```

Comment:
Can we leave `parts` passed not by const ref (we pass it as rvalue anyway for SELECTs) and use `parts = ` here?